models/ActiveRecord.php

```diff
@@ -335,14 +335,44 @@ public function update()
     // Eliminar un register por su ID
     public function delete()
     {
-        $query = "DELETE FROM "  . static::$tableName . " WHERE id = " . self::$database->escape_string(self::$primaryKey);
+        
+        $query = "DELETE FROM "  . static::$tableName . " WHERE id = " . $this->id;
         $result = self::$database->query($query);
         return $result;
     }
 
     public static function deleteAll(){
-        $query = "DROP ". static::$tableName;
+       // Borra todos los registros de la tabla
+        $query = "DELETE FROM " . static::$tableName;
+        $result1 = self::$database->query($query);
+
+        // Resetea el contador de autoincremento
+        $result2 = self::resetAutoIncrement();
+
+        if ($result1 && $result2) {
+            return true; // Borrado y reinicio exitoso
+        } else {
+            return false; // Hubo un error en al menos una de las consultas
+        }
```

Comment:
```suggestion
       return $result1 && $result2;
```
Como no estas realizando ninguna acción mas, es mejor retornar el resultado de la expresión en lugar de un if else